### PR TITLE
Spectron testing suite

### PR DESCRIPTION
**Pull request by evertonfraga**
This PR fixes the [spectron](https://github.com/electron/spectron) test suite, and adds a bunch of tests.

- Fixes the broken build.
- It creates a separate, clean, LokiDB for the tests.
- Uses [Client Binaries](https://github.com/ethereum/ethereum-client-binaries) in order to run `geth` on CI server
- Tests covers Mist sec issues.

ongoing: 
- Travis CI configuration.

More info: https://github.com/ethereum/mist/projects/2

### CODE REVIEW

**Comment by evertonfraga** [comment]
running tests.
image

File: tests/_base.js [hunk 1]
```diff
@@ -85,49 +92,73 @@ exports.mocha = function (_module, options) {
                     rpcport: 58545,
                 },
             });
-
+            console.log(this.geth);
```
Comment (frozeman) [inline tests/_base.js]: This should be removed

File: tests/_base.js [hunk 2]
```diff
@@ -85,49 +92,73 @@ exports.mocha = function (_module, options) {
...
             yield this.geth.start();
 
             this.web3 = new Web3(new Web3.providers.HttpProvider('http://localhost:58545'));
```
Comment (frozeman) [inline tests/_base.js]: The port is chosen because of a custom geth one?
Why using an HTTP provider here?

**Comment by evertonfraga** [comment]
(closing and reopening PR to try to trigger Travis CI)

**Pull request merged**